app/controllers/cart_controller.rb

```diff
@@ -33,4 +70,13 @@ def logged_in_user
     store_location
     redirect_to login_url
   end
+
+  def total_price
+    @total_price = 0
+    @cart = session[:cart] || {}
+    @cart.each do |key, value|
+      product = Product.find_by id: key
+      @total_price += product.price * value
```

Comment:
em có 2 hướng
nếu muốn kỹ: cần check xem product id trong hassh (tất cả các key trong hash) xem có cái nào ko tồn tại trong DB ko? nếu có báo message bảo reload page 

hoặc đơn giản ở đây, trc khi tính price* value, em check xem product có tồn tại ko?
nếu có thì tính toán, nếu ko thì remove cái product đó trong hash session đi

chị thích cái thứ 2